apis/refs/v1beta1/projectref.go

```diff
@@ -168,3 +168,24 @@ func ResolveProjectID(ctx context.Context, reader client.Reader, obj *unstructur
 
 	return "", fmt.Errorf("cannot find project id for %v %v/%v", obj.GetKind(), obj.GetNamespace(), obj.GetName())
 }
+
+type ProjectIDAndNumer struct {
+	ID     string
+	Number string
```

Comment:
Ideally this ProjectNumber should be int64. But then the mockgcp is overly complicated because we want to normalize the ProjectNumber int64 to `${projectNumber}`, but this break the actual reconciliation because it parses`${projectNumber}`  to int64.